protostar/start.py

```diff
@@ -37,8 +38,28 @@ def parse_args(parser: ArgumentParserFacade) -> Any:
         sys.exit(1)
 
 
-def run_protostar(protostar_cli: ProtostarCLI, parser: ArgumentParserFacade, args: Any):
+def run_protostar(
+    protostar_cli: ProtostarCLI,
+    parser: ArgumentParserFacade,
+    args: Any,
+):
     try:
+        if hasattr(args, "inputs") and isinstance(args.inputs, dict):
+            abi = asyncio.run(abi_from_args(args))
+            target_abi_item = None
+            for abi_item in abi:
+                if abi_item["name"] == args.function:
+                    target_abi_item = abi_item
+                    break
+            assert target_abi_item
+            inputs_as_list = []
+            for input_item in target_abi_item["inputs"]:
+                new_item = args.inputs[input_item["name"]]
+                if isinstance(new_item, list):
+                    inputs_as_list += new_item
+                else:
+                    inputs_as_list.append(new_item)
+            args.inputs = inputs_as_list
```

Comment:
This code definitely shouldn't be in the `start.py`.

Reply:
> I realize this code's not perfect and there's some refactoring ahead but I'm putting this up to just present the idea for this.